packages/shorebird_cli/lib/src/commands/patch/patch_android_command.dart

```diff
@@ -78,6 +78,10 @@ If this option is not provided, the version number will be determined from the p
         'staging',
         negatable: false,
         help: 'Whether to publish the patch to the staging environment.',
+      )
+      ..addOption(
+        'codesign-private-key',
+        hide: true,
```

Comment:
I don't think the final product will ever work this way (where it takes a private key and does the signing).

I could imagine it taking a key store (which is what gradle does).

But I think more likely it's going to take some sort of option (probably as part of shorebird.yaml), of a way to delegate to either a web-based signing service, or a command-line based one.

It's also possible that we will want to split our patch creation into two phases.  One which creates the artifacts, and a second which uploads them.  That's probably somewhat error prone (might be a non-trivial amount of state that has to be carefully passed to the "upload" phase), but may end up needed, depending on how long we expect signing to take (it might require human intervention).

Another possibility would be for us to do an entire `patch` release and have it in an "unsigned" state (similar to the current not-live state) and have it need to transition to a signed state before it can be made live?

Reply:
One could even imagine that we want patches to go through some sort of state machine on the server.  Where they start out in some "new" state, transition to "uploaded", "signed", "live", etc. 🤷

Reply:
Regardless of my comments above, if this unblocks demos, you should go forward with it. Just know that this isn't likely to be what you end up with for an architecture.

Reply:
Oh, for sure, this option in the CLI is just a "less hardecoded" way for us to do the the demo for step 1 and move forward with the other parts of the full feature

Reply:
Solving given the discussion (both here and the offline one) and since I proposed a different approach in the, now ready to review, PR